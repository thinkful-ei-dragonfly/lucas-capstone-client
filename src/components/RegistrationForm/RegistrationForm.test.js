import React from 'react'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

import RegistrationForm from './RegistrationForm'
configure({ adapter: new Adapter() });
describe('RegistrationForm component', () => {
  const props = {
    something: 'something'
  }

  it('renders a Component by default', () => {
    const wrapper = shallow(<RegistrationForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders a form', () => {
    const form = shallow(<RegistrationForm/>)
    expect(toJson(form)).toMatchSnapshot()
  })
  it('renders a full_name label in the form', () => {
    const full_name = shallow(<RegistrationForm />)
      .find('.full_name')
    expect(toJson(full_name)).toMatchSnapshot()
  })
  it('renders a user_name label in the form', () => {
    const user_name = shallow(<RegistrationForm />)
      .find('.user_name')
    expect(toJson(user_name)).toMatchSnapshot()
  })
  it('renders a password label in the form', () => {
    const password = shallow(<RegistrationForm />)
      .find('.password')
    expect(toJson(password)).toMatchSnapshot()
  })
})
