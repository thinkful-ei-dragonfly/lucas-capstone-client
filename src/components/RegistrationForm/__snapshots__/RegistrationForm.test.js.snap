// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders a Component by default 1`] = `
<form
  aria-label="Registration Form"
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="full_name"
  >
    <label
      htmlFor="RegistrationForm__full_name"
    >
      Full name
    </label>
    <input
      aria-label="Full Name"
      id="RegistrationForm__full_name"
      name="full_name"
      placeholder="Lorem Ipsum"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      User name
    </label>
    <input
      aria-label="Username"
      id="RegistrationForm__user_name"
      name="user_name"
      placeholder="loremipsum"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password
    </label>
    <input
      aria-label="password"
      id="RegistrationForm__password"
      name="password"
      placeholder="ultra super incredibly secure password"
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Register
  </button>
</form>
`;

exports[`RegistrationForm component renders a form 1`] = `
<form
  aria-label="Registration Form"
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="full_name"
  >
    <label
      htmlFor="RegistrationForm__full_name"
    >
      Full name
    </label>
    <input
      aria-label="Full Name"
      id="RegistrationForm__full_name"
      name="full_name"
      placeholder="Lorem Ipsum"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      User name
    </label>
    <input
      aria-label="Username"
      id="RegistrationForm__user_name"
      name="user_name"
      placeholder="loremipsum"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password
    </label>
    <input
      aria-label="password"
      id="RegistrationForm__password"
      name="password"
      placeholder="ultra super incredibly secure password"
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Register
  </button>
</form>
`;

exports[`RegistrationForm component renders a full_name label in the form 1`] = `
<div
  className="full_name"
>
  <label
    htmlFor="RegistrationForm__full_name"
  >
    Full name
  </label>
  <input
    aria-label="Full Name"
    id="RegistrationForm__full_name"
    name="full_name"
    placeholder="Lorem Ipsum"
    required={true}
    type="text"
  />
</div>
`;

exports[`RegistrationForm component renders a password label in the form 1`] = `
<div
  className="password"
>
  <label
    htmlFor="RegistrationForm__password"
  >
    Password
  </label>
  <input
    aria-label="password"
    id="RegistrationForm__password"
    name="password"
    placeholder="ultra super incredibly secure password"
    required={true}
    type="password"
  />
</div>
`;

exports[`RegistrationForm component renders a user_name label in the form 1`] = `
<div
  className="user_name"
>
  <label
    htmlFor="RegistrationForm__user_name"
  >
    User name
  </label>
  <input
    aria-label="Username"
    id="RegistrationForm__user_name"
    name="user_name"
    placeholder="loremipsum"
    required={true}
    type="text"
  />
</div>
`;
